### PR TITLE
Rework checks for repos

### PR DESCRIPTION
**Pull request by vuntz**
The checks are a bit more solid:
 - we don't always check for the product, but also for some other tag
 - we also check for the key signing the repo
 - there are more checks for SLE12

This comes on top of https://github.com/crowbar/crowbar/pull/2080 (so patches from this PR are included here for now).

### CODE REVIEW

File: releases/development/master/extra/install-chef-suse.sh [hunk 1]
```diff
@@ -549,53 +596,99 @@ check_media_links () {
     fi
 }
 
-check_media_links $MEDIA
+# Automatically create symlinks for SMT-mirrored repos if they exist
+for repo in SLES11-SP3-Pool \
+            SLES11-SP3-Updates \
+	    SUSE-Cloud-5-Pool \
```
Comment (tboerger) [inline releases/development/master/extra/install-chef-suse.sh]: Indention :)

**Comment by tboerger** [comment]
How can we prepare install-chef-suse.sh to be able to install openSUSE as well? :)

**Comment by vuntz** [comment]
> How can we prepare install-chef-suse.sh to be able to install openSUSE as well? :)

We can probably split the script to use some helpers, and then the repo checks could become distro-specific. But that's a topic for a later PR, I'd say :-)

**Comment by tboerger** [comment]
> We can probably split the script to use some helpers, and then the repo checks could become distro-specific. But that's a topic for a later PR, I'd say :-)

Sure, i just wanted to ask :)

File: releases/development/master/extra/install-chef-suse.sh [hunk 2]
```diff
@@ -500,47 +524,70 @@ check_repo_content () {
...
+        xml=$repo_dir/repodata/repomd.xml
+        ;;
+      *)
+        die "Internal error: unknown tag $tag for check_repo_tag"
+        ;;
+     esac
 
-MEDIA=/srv/tftpboot/suse-11.3/install
+    ignore_failure=1
+    skip_check_for_repo "$repo" || ignore_failure=0
```
Comment (jsuchome) [inline releases/development/master/extra/install-chef-suse.sh]: Isn't this reversed? It seems that if $repo **is** marked for check skip, than you may hit that ``die`` later

Comment (vuntz) [inline releases/development/master/extra/install-chef-suse.sh]: It's just that I use `0` and `1` as booleans, not as `shell exit codes interpreted for tests`. So I think it's correct, although possibly confusing.

**Comment by jsuchome** [comment]
+1

**Comment by aspiers** [comment]
Thanks a lot for these much-needed improvements!  Does this fix [bsc#838125](https://bugzilla.suse.com/show_bug.cgi?id=838125)?  There's also [bsc#910667](https://bugzilla.suse.com/show_bug.cgi?id=910667) which still needs fixing.

**Comment by aspiers** [comment]
Obsoletes #2048?

**Comment by vuntz** [comment]
@aspiers I (hopefully) updated code for your comments.

**Pull request merged**